### DISCUSSION TITLE
Data reference changes twice with useMutation

### DISCUSSION THREAD

**Discussion by fiddep** [TanStack/query#323] [General] [answered]
Hi!

I have this problem when using useMutation.
When calling the provided function the reference to the data changes twice.
The first time without anything added to the array of data. The second time a entry is added.
The second time is given since it's a new array. But can't figure out why the reference changes the first time.

Example of the mutation 
```
  const [createMutate] = useMutation(materialApi.post, {
    onSuccess: (x) => queryCache.setQueryData(queryId, xs => [...xs, x]),
  });
```

Am I missing something or is this behavior intended?

**Comment by cherniavskii** [comment]
Can you provide a reproducible example?

**Comment by fiddep** [comment, marked as answer]
Turns out that it was in combination with profiling the page.
Jumping between chrome devtools and the page causes the fetch on window focus to run, which was complicating the profiling.

**Comment by tannerlinsley** [comment]
Ah yep. Window refocus refetching is on by default, so you'll need to take that into account when debugging.
